components/scream/src/physics/p3/atmosphere_microphysics.hpp

```diff
@@ -79,13 +79,21 @@ class P3Microphysics : public AtmosphereProcess
         // Exner
         const Spack opmid(pmid(icol,ipack));
         const Smask opmid_mask(!isnan(opmid) and opmid>0.0);
```

Comment:
We can remove the isnan check. `opmid>0.0` is already false for entries that are nan, since every nan compares false to anything,regardless of the rhs and regardless of whether the comparison is ==,>,<,<=, or >=.

Reply:
On a side note: is it even possible that pmid<=0 ?

Reply:
I suppose a better question for these, and other similar mask declarations is, would it make sense to just go with the `X<0` which will catch the NaNs too as you mentioned.  Even if the code should have crashed by now because `X<0` is totally non-physical?  Or should we stick with the `isnan` check which should mask padded values in most cases and of course mask any NaNs that pop up?

Reply:
I'm not sure I understand what you are suggesting. X<0 (like X<=0, X==0, X>0, X>=0) will be false for NaNs, btw. What I was suggesting is to simply do `m = X>0;`, since `m[i]` will be false if `X[i]<=0` as well as if `X[i]` is NaN, hence doing two checks in one.

Reply:
That said, we should not check for NaNs, I think. Otherwise, where to we draw the line? Should we always check every single input for NaNs? If your goal was to simply mask out the padding at the end, then my suggestion is to not do it. We don't care what's in the padding, so even if it's NaNs, that's fine.

Reply:
Yes, it can. Creating a `Smask(true)` is pointless. Just do `X(icol,ilev) = PF::compute_blah(...)`.